### PR TITLE
Implement support for user supplied ajv instances

### PR DESCRIPTION
**Pull request by MSE99**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

#### Description

Implements requested feature in #72 , with this PR the users can supply their own instance of ajv, i originally wanted to implement the `current` keyword too but the issue was not clear on what the keyword does or how it should be implemented.

### CODE REVIEW

**Comment by mcollina** [comment]
Could you please add a test for the new type: https://github.com/fastify/env-schema/blob/master/test/types/types.test.ts?

**Review by mcollina — APPROVED** [review]
lgtm

File: index.js [hunk 1]
```diff
@@ -41,11 +41,12 @@ const optsSchema = {
     },
     env: { type: 'boolean', default: true },
     dotenv: { type: ['boolean', 'object'], default: false },
-    expandEnv: { type: ['boolean'], default: false }
+    expandEnv: { type: ['boolean'], default: false },
+    ajv: { type: 'object' }
```
Comment (Eomm) [inline index.js:45]:
```suggestion
    ajv: { type: 'object', additionalProperties: true }
```

File: index.d.ts [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import Ajv from "ajv";
```
Comment (Eomm) [inline index.d.ts:1]: Could TS users that add ajv7 instance have trouble with this interface that it is ajv6?

**Review by climba03003 — COMMENTED** [review]
Can you expose the `seperator` plugin, it maybe useful to the users.

File: index.d.ts (hunk 2 above)
Comment (MSE99) [reply index.d.ts:1]: Yeah they're going to have trouble, ajv 8 removes some properties from the options object and adds new ones the options object type between version 6, 7 and 8 are similar but is not compatible I'm not entirely sure on how to deal with this issue.

**Review by climba03003 — APPROVED** [review]
LGTM.

**Pull request merged**